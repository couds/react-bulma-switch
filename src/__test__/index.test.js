import React from 'react';
import renderer from 'react-test-renderer';
import Switch from '..';

describe('Switch component', () => {
  it('Should Exist', () => {
    expect(Switch).toMatchSnapshot();
  });
  it('Should have Switch classname', () => {
    const component = renderer.create(<Switch id="test">Facebook</Switch>);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should concat Bulma class with classes in props', () => {
    const component = renderer.create(<Switch id="test" className="other-class test">Facebook</Switch>);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should create random id if no id passed', () => {
    const component = renderer.create(<Switch>Switch 1</Switch>);
    const component2 = renderer.create(<Switch>Switch 2</Switch>);
    expect(component.getInstance().id).not.toEqual(component2.getInstance().id);
  });
});
