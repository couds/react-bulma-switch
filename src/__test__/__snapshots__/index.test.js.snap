// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch component Should Exist 1`] = `[Function]`;

exports[`Switch component Should concat Bulma class with classes in props 1`] = `
<div
  className="field other-class test"
  style={Object {}}
>
  <input
    checked={false}
    className="switch"
    disabled={false}
    id="test"
    type="checkbox"
    value={false}
  />
  <label
    className="label"
    htmlFor="test"
    style={Object {}}
  >
    Facebook
  </label>
</div>
`;

exports[`Switch component Should have Switch classname 1`] = `
<div
  className="field"
  style={Object {}}
>
  <input
    checked={false}
    className="switch"
    disabled={false}
    id="test"
    type="checkbox"
    value={false}
  />
  <label
    className="label"
    htmlFor="test"
    style={Object {}}
  >
    Facebook
  </label>
</div>
`;
